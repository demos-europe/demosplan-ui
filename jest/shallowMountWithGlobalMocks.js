/**
 * Used to run Jest-Tests
 */

// Use Local Vue for testing
import { shallowMount } from '@vue/test-utils'

// then you can use `app.component` here
// Mocking global stuff
const globalMocks = {
  hasPermission: jest.fn(() => true),
  Translator: {
    trans: jest.fn(key => key)
  }
}

/**
 * A shallowMount for the instance but with some additional Mocks
 * So we don't have to do that in every Test.
 *
 * @param component{Component}
 * @param options{mocks, props, computed, store, methods, stubs, slots}
 *
 * @returns {Vue}
 */
const shallowMountWithGlobalMocks = (component, options) => {
  return shallowMount(component, {
      ...options,
      global: {
        mocks: globalMocks,
      }
    }
  )
}

export default shallowMountWithGlobalMocks
