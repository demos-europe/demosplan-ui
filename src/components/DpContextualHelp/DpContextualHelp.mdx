import { Canvas, Meta } from '@storybook/blocks'
import * as DpContextualHelpStories from './DpContextualHelp.stories.tsx'

<Meta of={DpContextualHelpStories} />

# Contextual Help

The Contextual Help component renders an icon indicating contextual help,
which shows a tooltip able to handle text or HTML content.

## Default usage

<Canvas of={DpContextualHelpStories.Default} />

## With custom icon

<Canvas of={DpContextualHelpStories.CustomIcon} />

## With custom size

<Canvas of={DpContextualHelpStories.CustomSize} />

## With tooltip options

<Canvas of={DpContextualHelpStories.TooltipOptions} />