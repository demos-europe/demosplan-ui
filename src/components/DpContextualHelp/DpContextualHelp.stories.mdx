# Contextual Help

The Contextual Help component renders an icon indicating contextual help,
which shows a tooltip able to handle text or html content.

## Default usage

```html
<dp-contextual-help :text="Translator.trans('statements.import.hint')" />
```
