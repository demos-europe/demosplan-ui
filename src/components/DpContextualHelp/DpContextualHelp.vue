<template>
  <dp-icon
    :icon="icon"
    :aria-label="ariaLabel"
    v-tooltip="text" />
</template>

<script>
import { de } from '../shared/translations'
import DpIcon from '../DpIcon/DpIcon'
import { SIZES } from '../DpIcon/util/iconVariables'
import { Tooltip } from '../../directives'

export default {
  name: 'DpContextualHelp',

  components: {
    DpIcon
  },

  props: {
    /**
     * The icon displayed as trigger for the tooltip.
     */
    icon: {
      type: String,
      required: false,
      default: 'question'
    },

    /**
     * The icon size. May be small, medium, or large.
     */
    size: {
      type: String,
      required: false,
      default: 'medium',
      validator: (prop) => Object.keys(SIZES).includes(prop)
    },

    /**
     * A string representing the actual tooltip content. May include html.
     */
    text: {
      type: String,
      required: true
    }
  },

  directives: {
    tooltip: Tooltip
  },

  data () {
    return {
      ariaLabel: de.contextualHelp
    }
  }
}
</script>
