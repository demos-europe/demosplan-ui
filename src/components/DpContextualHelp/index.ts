export { default } from './DpContextualHelp.vue'
export * from './DpContextualHelp.vue'
