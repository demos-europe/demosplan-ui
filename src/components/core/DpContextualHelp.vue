<script>
import Tooltip from './../../directives/Tooltip/Tooltip'

export default {
  name: 'DpContextualHelp',

  functional: true,

  props: {
    /**
     * A translation key representing the actual tooltip content.
     * May actually be any string including html.
     */
    text: {
      type: String,
      required: true
    }
  },

  directives: {
    tootip: Tooltip
  },

  render: function (h, { props, data }) {
    return h('i', {
      attrs: {
        class: `fa fa-question-circle ${typeof data.staticClass === 'undefined' ? '' : data.staticClass}`,
        'aria-label': Translator.trans('contextual.help')
      },
      directives: [
        {
          name: 'tooltip',
          value: props.text
        }
      ]
    })
  }
}
</script>
