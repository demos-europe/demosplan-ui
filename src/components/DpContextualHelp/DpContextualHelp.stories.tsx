import type { Meta, StoryObj } from '@storybook/vue'

import DpContextualHelp from './DpContextualHelp.vue'

const Translator = {
  trans: (key: string): string => key
}

const meta: Meta<typeof DpContextualHelp> = {
  component: DpContextualHelp,
  title: "Components/DpContextualHelp",
}

export default meta
type Story = StoryObj<typeof DpContextualHelp>

export const Default: Story = {
  args: {
    text: 'test text'
  }
}
