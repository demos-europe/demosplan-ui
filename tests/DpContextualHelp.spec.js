/**
 * (c) 2010-present DEMOS E-Partizipation GmbH.
 *
 * This file is part of the package demosplan,
 * for more information see the license file.
 *
 * All rights reserved
 */

import DpContextualHelp from '../src/components/DpContextualHelp/DpContextualHelp.vue'
import shallowMountWithGlobalMocks from '../jest/shallowMountWithGlobalMocks'

describe('DpContextualHelp', () => {
  it('should be an object', () => {
    expect(typeof DpContextualHelp).toBe('object')
  })

  it('should be named DpContextualHelp', () => {
    expect(DpContextualHelp.name).toBe('DpContextualHelp')
  })

  it('should render the correct html', async () => {

    const instance = shallowMountWithGlobalMocks(DpContextualHelp, {
      propsData: {
        text: 'This is the tooltip content.'
      }
    })

    expect(instance.html()).toMatchSnapshot()
  })
})
