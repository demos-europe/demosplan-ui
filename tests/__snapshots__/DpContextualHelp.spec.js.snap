// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DpContextualHelp should render the correct html 1`] = `"<dp-icon-stub icon="question" size="medium" aria-label="Kontexthilfe"></dp-icon-stub>"`;
